src/scripts/clipperUI/components/modeButtonSelector.tsx

```diff
@@ -31,101 +25,140 @@ class ModeButtonSelectorClass extends ComponentBase<{}, ClipperStateProp> {
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.ScreenShot.Button.Tooltip")
+		};
 	}
 
-	private getRegionModeButton(currentMode: ClipMode) {
+	private getRegionButtonProps(currentMode: ClipMode): PropsForModeElementNoAriaGrouping {
 		let enableRegionClipping = this.props.clipperState.injectOptions && this.props.clipperState.injectOptions.enableRegionClipping;
 		let contextImageModeUsed = this.props.clipperState.invokeOptions && this.props.clipperState.invokeOptions.invokeMode === InvokeMode.ContextImage;
 
 		if (!enableRegionClipping && !contextImageModeUsed) {
 			return undefined;
 		}
 
-		return <ModeButton imgSrc={ExtensionUtils.getImageResourceUrl("region.png") }
-			label={Localization.getLocalizedString(this.getRegionModeButtonLabel())}
-			myMode={ClipMode.Region} tabIndex={41} selected={currentMode === ClipMode.Region}
-			onModeSelected={this.onModeSelected.bind(this) }
-			tooltipText={Localization.getLocalizedString("WebClipper.ClipType.MultipleRegions.Button.Tooltip")}/>;
+		return {
+			imgSrc: ExtensionUtils.getImageResourceUrl("region.png"),
+			label: Localization.getLocalizedString(this.getRegionButtonLabel()),
+			myMode: ClipMode.Region,
+			selected: currentMode === ClipMode.Region,
+			onModeSelected: this.onModeSelected.bind(this),
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.MultipleRegions.Button.Tooltip")
+		};
 	}
 
-	private getRegionModeButtonLabel(): string {
+	private getRegionButtonLabel(): string {
 		return "WebClipper.ClipType.Region.Button";
 	}
 
-	private getSelectionModeButton(currentMode: ClipMode) {
+	private getSelectionButtonProps(currentMode: ClipMode): PropsForModeElementNoAriaGrouping {
 		if (this.props.clipperState.invokeOptions.invokeMode !== InvokeMode.ContextTextSelection) {
 			return undefined;
 		}
 
-		return <ModeButton imgSrc={ExtensionUtils.getImageResourceUrl("select.png") }
-			label={Localization.getLocalizedString("WebClipper.ClipType.Selection.Button")}
-			myMode={ClipMode.Selection} tabIndex={43} selected={currentMode === ClipMode.Selection}
-			onModeSelected={this.onModeSelected.bind(this) }
-			tooltipText={Localization.getLocalizedString("WebClipper.ClipType.Selection.Button.Tooltip")}/>;
+		return {
+			imgSrc: ExtensionUtils.getImageResourceUrl("select.png"),
+			label: Localization.getLocalizedString("WebClipper.ClipType.Selection.Button"),
+			myMode: ClipMode.Selection,
+			selected: currentMode === ClipMode.Selection,
+			onModeSelected: this.onModeSelected.bind(this),
+			tooltipText: Localization.getLocalizedString("WebClipper.ClipType.Selection.Button.Tooltip")
+		};
 	}
 
-	private getBookmarkModeButton(currentMode: ClipMode) {
-		if (this.props.clipperState.pageInfo.rawUrl.indexOf("file:///") === 0) {
+	private getBookmarkButtonProps(currentMode: ClipMode): PropsForModeElementNoAriaGrouping {
+		if (this.props.clipperState.pageInfo.rawUrl.slice(0, 8) === "file:///") {
```

Comment:
On hindsight, I still think indexOf is better simply because it's safer to use (despite probably being less efficient). Right now if the slice length and the string length and mismatched, if will always return false. It's an easy bug to introduce.